### ISSUE TITLE
Make prefetching work with Run and LuminosityBlock transitions

### ISSUE THREAD

**Issue by Dr15Jones** [cms-sw/cmssw#21906]
Right now prefetching only works with Events. We need it to work with the other transitions as well so that the framework can schedule running modules based on dependencies.

**Comment by Dr15Jones** [comment]
assign core

**Comment by wddgit** [comment]
I reviewed the code related to prefetching and it looks as though Chris had already implemented this prefetching and forgot that he had done it. From a one pass read through of the code, it looks good already. Note that there are two functions that call doWorkNoPrefetchingAsync, but these are only used for stream transitions and since we do not put products into the Run or LuminosityBlock on stream transitions this is OK. There cannot be any data dependencies in stream transitions for runs and lumis so we do not need prefetching. For global transitions, we are properly calling doWorkAsync which does prefetching. I added some comments into the code in PR #21950 to clarify this so we remember next we read this code. This issue can be closed.
